src/providers/data_provider/dp_target_id.c

```diff
@@ -448,10 +448,238 @@ static void dp_req_initgr_pp_set_initgr_timestamp(struct dp_initgr_ctx *ctx,
     }
 }
 
+
+struct dp_sr_resolve_groups_state {
+    struct data_provider *provider;
+    struct dp_initgr_ctx *initgroups_ctx;
+    struct dp_reply_std reply;
+
+    uint32_t *resolve_gids; /* Groups needing resolution */
+    int resolve_gnum;
+    int num_iter;
+    uint32_t gnum;
+};
+
+static errno_t dp_sr_resolve_groups_check(struct dp_sr_resolve_groups_state *state);
+static errno_t dp_sr_resolve_groups_next(struct tevent_req *req);
+static void dp_sr_resolve_groups_done(struct tevent_req *subreq);
+
+struct tevent_req *
+dp_sr_resolve_groups_send(TALLOC_CTX *mem_ctx,
+                          struct tevent_context *ev,
+                          struct dp_reply_std reply,
+                          struct data_provider *provider,
+                          struct dp_initgr_ctx *initgr_ctx)
+{
+
+    struct dp_sr_resolve_groups_state *state;
+    struct tevent_req *req;
+    int ret;
```

Comment:
errno_t

Reply:
`errno_t` is a typedef for `int` and isn't used consistently:
```
$ grep -rn "errno_t ret" * | wc -l
2843

$ grep -rn "int ret" * | wc -l
1819
```

Btw: https://pagure.io/SSSD/sssd/issue/4027

But since `dp_sr_resolve_groups_*()` both return `errno_t` it would be cleaner to use `errno_t`.

Btw, `tevent_req_error(req, ret)`:
```
bool | tevent_req_error (struct tevent_req *req, uint64_t error)
```
 - implicit signed->unsigned cast. But quite a pattern in the code base.